File: app/src/main/java/com/example/adiphong/rx2/OneFragment.java
package com.example.adiphong.rx2;

import android.app.Activity;
import android.content.Context;
import android.os.Build;
import android.os.Bundle;
import android.support.annotation.RequiresApi;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.inputmethod.InputMethodManager;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ScrollView;

public class OneFragment extends Fragment {

    private MainActivity mainActivity;

    public OneFragment() {
        // Required empty public constructor
    }

    public static OneFragment newInstance() {
        OneFragment fragment = new OneFragment();
        Bundle args = new Bundle();
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

    }

    @RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN)
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view = inflater.inflate(R.layout.fragment_one, container, false);

        Button button = view.findViewById(R.id.next_button);

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                BlankFragment blankFragment = BlankFragment.newInstance();
                String tag = "blankFragment";
                mainActivity.replaceFragment(blankFragment, tag);
            }
        });
        return view;
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);

        if (getActivity() instanceof MainActivity) {
            mainActivity = (MainActivity) getActivity();
        }
    }
}

File: app/src/main/java/com/example/adiphong/rx2/OneFragment.kt
package com.example.adiphong.rx2

import android.content.Context
import android.os.Bundle
import android.support.v4.app.Fragment
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import io.reactivex.Observable
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers
import java.sql.DriverManager.println
import java.util.concurrent.TimeUnit

class OneFragment : Fragment() {

    private var mainActivity: MainActivity? = null

    private var loadingViewDialog: LoadingViewDialog? = null

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        val view = inflater!!.inflate(R.layout.fragment_one, container, false)

        loadingViewDialog = LoadingViewDialog(activity)

        val button = view.findViewById<Button>(R.id.next_button)

        button.setOnClickListener {
            val blankFragment = BlankFragment.newInstance()
            val tag = "blankFragment"
            mainActivity!!.replaceFragment(blankFragment, tag)
        }
        return view
    }

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        println("------------------------ onActivityCreated")

//        Observable.just(randomInteger)
//                .delay(10, TimeUnit.SECONDS)
//                .subscribeOn(Schedulers.io())
//                .observeOn(AndroidSchedulers.mainThread())
//                .doOnSubscribe { __ -> loadingViewDialog!!.showLoading() }
//                .doFinally { loadingViewDialog!!.dismissLoading() }
//                .subscribe { longTimed -> Log.d("---------- tag", "&&&& on timer") }

//        Observable.create<Int> { emitter ->
//            if (!emitter.isDisposed) {
//                val randomInteger = (1..100).shuffled().last()
//                emitter.onNext(randomInteger)
//                println("------------------ randomInteger $randomInteger")
//
//            }
//        }
//                .repeatWhen { t ->
//                    println("------------------------ repeatWhen $t ")
//                    t.delay(5, TimeUnit.SECONDS)
//                }
//                Observable.just((1..100).shuffled().last())
//                                        .repeatWhen { t ->
//                    println("------------------------ repeatWhen $t ")
//                    t.delay(5, TimeUnit.SECONDS)
//                }
//                .takeUntil { t: Int ->
//                    println("------------------------ takeUntil $t ")
//                    t == 10
//                }
//                .subscribeOn(Schedulers.io())
//                .observeOn(AndroidSchedulers.mainThread())
//                .doOnSubscribe { _ -> loadingViewDialog!!.showLoading() }
//                .doFinally {
//                    loadingViewDialog!!.dismissLoading()
//                }
//                .subscribe { longTimed ->
//                    Log.d("---------- tag", "finish")
//                }

    }

    override fun onAttach(context: Context?) {
        super.onAttach(context)

        if (activity is MainActivity) {
            mainActivity = activity as MainActivity
        }
    }

    companion object {

        fun newInstance(): OneFragment {
            val fragment = OneFragment()
            val args = Bundle()
            fragment.arguments = args
            return fragment
        }
    }
}
